db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/CreateIndexNode.java

```diff
@@ -140,170 +150,262 @@ public void init(
+                        columnDescriptor.getType().getTypeId().getSQLTypeName());
+            }
+        }
+
+        /* Check for number of key columns to be less than 16 to match DB2 */
+/*        if (columnCount > 16)
+            throw StandardException.newException(SQLState.LANG_TOO_MANY_INDEX_KEY_COLS);
 */
-		/* See if the index already exists in this schema.
-		 * NOTE: We still need to check at execution time
-		 * since the index name is only unique to the schema,
-		 * not the table.
-		 */
-//  		if (dd.getConglomerateDescriptor(indexName.getTableName(), sd, false) != null)
-//  		{
-//  			throw StandardException.newException(SQLState.LANG_OBJECT_ALREADY_EXISTS_IN_OBJECT,
-//  												 "Index",
-//  												 indexName.getTableName(),
-//  												 "schema",
-//  												 sd.getSchemaName());
-//  		}
-
-		/* Statement is dependent on the TableDescriptor */
-		getCompilerContext().createDependency(td);
-
-	}
-
-	/**
-	 * Return true if the node references SESSION schema tables (temporary or permanent)
-	 *
-	 * @return	true if references SESSION schema tables, else false
-	 *
-	 * @exception StandardException		Thrown on error
-	 */
-	public boolean referencesSessionSchema()
-		throws StandardException
-	{
-		//If create index is on a SESSION schema table, then return true.
-		return isSessionSchema(td.getSchemaName());
-	}
-
-	/**
-	 * Create the Constant information that will drive the guts of Execution.
-	 *
-	 * @exception StandardException		Thrown on failure
-	 */
-	public ConstantAction	makeConstantAction() throws StandardException
-	{
-		SchemaDescriptor		sd = getSchemaDescriptor();
-
-		int columnCount = columnNames.length;
-		int approxLength = 0;
-		boolean index_has_long_column = false;
-
-
-		// bump the page size for the index,
-		// if the approximate sizes of the columns in the key are
-		// greater than the bump threshold.
-		// Ideally, we would want to have atleast 2 or 3 keys fit in one page
-		// With fix for beetle 5728, indexes on long types is not allowed
-		// so we do not have to consider key columns of long types
-        for (String columnName : columnNames) {
+        /* See if the index already exists in this schema.
+         * NOTE: We still need to check at execution time
+         * since the index name is only unique to the schema,
+         * not the table.
+         */
+//          if (dd.getConglomerateDescriptor(indexName.getTableName(), sd, false) != null)
+//          {
+//              throw StandardException.newException(SQLState.LANG_OBJECT_ALREADY_EXISTS_IN_OBJECT,
+//                                                   "Index",
+//                                                   indexName.getTableName(),
+//                                                   "schema",
+//                                                   sd.getSchemaName());
+//          }
+
+        if (onExpression) {
+            bindIndexExpressions();
+            generateExecutableIndexExpression();
```

Comment:
It seems more appropriate to put  generateExecutableIndexExpression() in makeConstantAction() at code generation phase.

Reply:
Right, that makes more sense. Fixed in new commit.